### ISSUE TITLE
Bug: using ExecuteQueryMultipleAsync with many resultsets in some cases results in a timeout 

### ISSUE THREAD

**Issue by Garios** [mikependon/RepoDb#1127]
### Bug Description

It's wired. I have 18 resultsets, which are extracted by using ExtractAsync. One of the last resultsets is about 50.000 records. Extracting it, results in a timeout. If I place this resultset on the first or second position, it runs without problem.

Do you have any idea, why this could happen?

**Library Version:**

Example: RepoDb v1.13.0 on MSSQL-Server

**Comment by Garios** [comment]
Hello,
In the meantime I thought the problem is on my side, but I made some tests and it seems that there is an issue with repodb. I also have another code-location, which also uses ExecuteQueryMultipleAsync. There are only three resultsets with a lot of records.

It's very confusing:
I have only one MSSQL-Database. The code in production uses repodb. It seems so, that sometime it comes to a slowdown on sql-server. This could happen for a few minutes. If I open my dev-system, which connects to the same sql-database with the same connection-string, I have the same behaviour. Accessing the data is very slow. If I increase the query, it results in a timeout.

But when I execute the query on Microsoft SQL Server Management Console, the query finishes in milliseconds.

Then I thought it could depend on SQL-Connection-String. Therefore I implemented all without repodb using casual way with DbDataReader. This runs extremely fast.

I also tried async and non-async.

I'm confused. What could happen here?
